lib/Sema/TypeCheckType.cpp

```diff
@@ -3453,6 +3472,13 @@ void TypeChecker::checkUnsupportedProtocolType(Decl *decl) {
       decl->getDeclContext()->isModuleScopeContext())
     return;
 
+  // Global variable declarations are okay to skip, because at the top level of script/repl
```

Comment:
Right, indirectly. The existing DeclChecker only calls this code (the decl version of checkUnsupportedProtocolType()) if it is doing a second pass or is in a protocol or extension context, and it only does a second pass over things at module context. So it isn't getting called anywhere where there could be statements, except for TopLevelCodeDecls when those are allowed.

But it doesn't skip nested decls. So talking through this has also led me to another couple cases of doubling (trebling, quadrupling...):
`protocol P {
    typealias T
}
protocol Q {
    var s: P { get set }
}` ... which will currently complain about protocol P four times, once for the PatternDecl, once for the VarDecl, and then once each for the anonymous get and set FuncDecls.

And: `extension Q {
    func bar(x: P) {}
}` ... which will complain twice, once for the top-level ExtensionDecl and once for the FuncDecl inside.

So clearly there is more to be done here. Thanks for the leading questions. :)